xbmc/video/VideoReferenceClock.cpp

```diff
@@ -94,18 +94,28 @@ void CVideoReferenceClock::Process()
   bool SetupSuccess = false;
   int64_t Now;
 
+  bool glx_fallback = false;
   while(!m_bStop)
   {
     //set up the vblank clock
 #if defined(HAVE_X11)
   std::string gpuvendor = g_Windowing.GetRenderVendor();
   std::transform(gpuvendor.begin(), gpuvendor.end(), gpuvendor.begin(), ::tolower);
-  if ((gpuvendor.compare(0, 5, "intel") == 0 ||
+  if (!glx_fallback && (gpuvendor.compare(0, 5, "intel") == 0 ||
        gpuvendor.compare(0, 5, "x.org") == 0)) // AMD
+  {
     m_pVideoSync = new CVideoSyncDRM();
 #if defined(HAS_GLX)
+    glx_fallback = true;
+#endif
+  }
+#if defined(HAS_GLX)
   else
+  {
+    // do not fallback twice
+    glx_fallback = false;
```

Comment:
You want an enum in the xbmc/video/VideoReferenceClock.h or is an integer okay, that I can just increase?